src/java.base/share/classes/java/util/zip/GZIPInputStream.java

```diff
@@ -95,8 +95,16 @@ public GZIPInputStream(InputStream in) throws IOException {
      * Reads uncompressed data into an array of bytes. If {@code len} is not
      * zero, the method will block until some input can be decompressed; otherwise,
      * no bytes are read and {@code 0} is returned.
+     * <p>
+     * If <i>n</i> denotes the returned number of inflated bytes then {@code buf[off]}
+     * through {@code buf[off+}<i>n</i>{@code -1]} will contain the uncompressed data. The
+     * elements {@code buf[off+}<i>n</i>{@code ]} through {@code buf[off+}<i>len</i>{@code -1]}
+     * are undefined and an implementation is free to change them during the inflate
+     * operation. If the return value is -1 or an exception is thrown the whole
```

Comment:
The sentence is very long. How about putting "an implementation is free ..." in parentheses after "undefined".